src/twisted/test/test_sslverify.py

```diff
@@ -288,17 +289,27 @@ def connectionLost(self, reason):
     plainServerFactory = protocol.Factory()
     plainServerFactory.protocol = lambda: serverWrappedProto
 
+    clock = Clock()
     clientFactory = TLSMemoryBIOFactory(
-        clientOpts, isClient=True, wrappedFactory=plainServerFactory
+        clientOpts, isClient=True, wrappedFactory=plainServerFactory, clock=clock
     )
     serverFactory = TLSMemoryBIOFactory(
-        serverOpts, isClient=False, wrappedFactory=plainClientFactory
+        serverOpts, isClient=False, wrappedFactory=plainClientFactory, clock=clock
     )
 
     sProto, cProto, pump = connectedServerAndClient(
         lambda: serverFactory.buildProtocol(None),
         lambda: clientFactory.buildProtocol(None),
     )
+
+    # Need time to pass for flushing to work:
```

Comment:
can we have this as the default behaviour for `pump` so that we don't have to make this change here.

I can see that things are complicated, since the previous code is not dealing with the clock.

I expect that with this change, many other projects for which the TLS tests are written using iosim might need to be updated.

If iosim `pump` will automatically handle the new buffering behaviour, no changes should be required in the test code of 3rd party projects.

so maybe just call something like `reactor.doIteration(0.000001)` inside `pump.flush()`

Maybe `reactor.doIteration(0)` will also work.